solidity/dashboard/src/pages/delegation/GrantedTokensPage.jsx

```diff
@@ -48,9 +45,11 @@ const GrantedTokensPageComponent = ({ onSubmitDelegateStakeForm }) => {
 
   useEffect(() => {
     if (!isEmptyArray(grants) && isEmptyObj(selectedGrant)) {
-      setSelectedGrant(grants[0])
+      const lookupGrantId = hash.substr(1)
+      const grantByHash = grants.find((grant) => grant.id === lookupGrantId)
+      setSelectedGrant(grantByHash || grants[0])
     }
-  }, [grants, selectedGrant, previousGrant.id])
+  }, [grants, selectedGrant, previousGrant.id, hash])
```

Comment:
Why put this in the hash and not make it part of the URL? e.g. `/delegation/grant/<id>`.